File: mobile-app/lib/models/missing_info_models.dart
/// 缺失信息分析结果
class MissingInfoAnalysisResult {
  final List<MissingInfoItem> missingInfo;
  final List<QuestionItem> questions;
  final bool hasCriticalMissing;

  MissingInfoAnalysisResult({
    required this.missingInfo,
    required this.questions,
    required this.hasCriticalMissing,
  });

  factory MissingInfoAnalysisResult.fromJson(Map<String, dynamic> json) {
    return MissingInfoAnalysisResult(
      missingInfo: (json['missingInfo'] as List?)
              ?.map((e) => MissingInfoItem.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
      questions: (json['questions'] as List?)
              ?.map((e) => QuestionItem.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
      hasCriticalMissing: json['hasCriticalMissing'] as bool? ?? false,
    );
  }
}

/// 缺失信息项
class MissingInfoItem {
  final String field;
  final String question;
  final String type;
  final bool required;
  final String? domain;
  final List<String>? options;
  final String? hint;

  MissingInfoItem({
    required this.field,
    required this.question,
    required this.type,
    required this.required,
    this.domain,
    this.options,
    this.hint,
  });

  factory MissingInfoItem.fromJson(Map<String, dynamic> json) {
    return MissingInfoItem(
      field: json['field'] as String,
      question: json['question'] as String,
      type: json['type'] as String,
      required: json['required'] as bool? ?? false,
      domain: json['domain'] as String?,
      options: (json['options'] as List?)?.map((e) => e as String).toList(),
      hint: json['hint'] as String?,
    );
  }
}

/// 问诊式问题项
class QuestionItem {
  final String questionId;
  final String question;
  final String type;
  final bool required;
  final List<String>? options;
  final String? hint;
  final String? field;

  QuestionItem({
    required this.questionId,
    required this.question,
    required this.type,
    required this.required,
    this.options,
    this.hint,
    this.field,
  });

  factory QuestionItem.fromJson(Map<String, dynamic> json) {
    return QuestionItem(
      questionId: json['questionId'] as String,
      question: json['question'] as String,
      type: json['type'] as String,
      required: json['required'] as bool? ?? false,
      options: (json['options'] as List?)?.map((e) => e as String).toList(),
      hint: json['hint'] as String?,
      field: json['field'] as String?,
    );
  }
}

/// 补全缺失信息请求
class CompleteMissingInfoRequest {
  final Map<String, dynamic> answers;
  final Map<String, List<String>>? attachments;

  CompleteMissingInfoRequest({
    required this.answers,
    this.attachments,
  });

  Map<String, dynamic> toJson() {
    return {
      'answers': answers,
      if (attachments != null) 'attachments': attachments,
    };
  }
}

File: backend/src/FieldTicket.Shared/Models/MissingInfoModels.cs
namespace FieldTicket.Shared.Models;

/// <summary>
/// 缺失信息项
/// </summary>
public class MissingInfoItem
{
    /// <summary>
    /// 字段名
    /// </summary>
    public string Field { get; set; } = string.Empty;
    
    /// <summary>
    /// 问题描述
    /// </summary>
    public string Question { get; set; } = string.Empty;
    
    /// <summary>
    /// 问题类型（yes_no/number/text/file/select）
    /// </summary>
    public string Type { get; set; } = "yes_no";
    
    /// <summary>
    /// 是否必填
    /// </summary>
    public bool Required { get; set; }
    
    /// <summary>
    /// 问题域
    /// </summary>
    public string? Domain { get; set; }
    
    /// <summary>
    /// 选项列表（用于select类型）
    /// </summary>
    public List<string>? Options { get; set; }
    
    /// <summary>
    /// 提示信息
    /// </summary>
    public string? Hint { get; set; }
}

/// <summary>
/// 问诊式问题项
/// </summary>
public class QuestionItem
{
    /// <summary>
    /// 问题ID
    /// </summary>
    public string QuestionId { get; set; } = string.Empty;
    
    /// <summary>
    /// 问题文本
    /// </summary>
    public string Question { get; set; } = string.Empty;
    
    /// <summary>
    /// 问题类型
    /// </summary>
    public string Type { get; set; } = "yes_no";
    
    /// <summary>
    /// 是否必填
    /// </summary>
    public bool Required { get; set; }
    
    /// <summary>
    /// 选项列表（用于select类型）
    /// </summary>
    public List<string>? Options { get; set; }
    
    /// <summary>
    /// 提示信息
    /// </summary>
    public string? Hint { get; set; }
    
    /// <summary>
    /// 关联的字段名
    /// </summary>
    public string? Field { get; set; }
}

/// <summary>
/// 补全缺失信息请求
/// </summary>
public class CompleteMissingInfoRequest
{
    /// <summary>
    /// 问题答案（QuestionId -> Answer）
    /// </summary>
    public Dictionary<string, object> Answers { get; set; } = new();
    
    /// <summary>
    /// 附件ID列表（用于文件类型问题）
    /// </summary>
    public Dictionary<string, List<Guid>> Attachments { get; set; } = new();
}

/// <summary>
/// 缺失信息分析结果
/// </summary>
public class MissingInfoAnalysisResult
{
    /// <summary>
    /// 缺失信息项列表
    /// </summary>
    public List<MissingInfoItem> MissingInfo { get; set; } = new();
    
    /// <summary>
    /// 问诊式问题清单
    /// </summary>
    public List<QuestionItem> Questions { get; set; } = new();
    
    /// <summary>
    /// 是否有关键信息缺失
    /// </summary>
    public bool HasCriticalMissing { get; set; }
}


